### ISSUE TITLE
Receiving messages not meant for me while updating desktop version

### ISSUE THREAD

**Issue by Janila** [WhisperSystems/Signal-Android#7151]
Sorry for not sticking to the proper bug report format, I'm a little scared and confused and wanted to get out a report as soon as possible.
tl;dr: my desktop version of signal is not showing my conversations but those of a friend.

The details:
I had Signal installed on my laptop (Ubuntu 16.04.2 LTS) as a chromium app and on my phone, everything working fine. I wanted to install the new standalone Signal on my laptop. Here are steps I did as well as I remember. Sorry, I don't know how to try to reproduce this, so there may also be irrelevant steps listed.

1. I followed the Linux install instructions:
	$ curl -s https://updates.signal.org/desktop/apt/keys.asc | sudo apt-key add -
	$ echo "deb [arch=amd64] https://updates.signal.org/desktop/apt xenial main" | sudo tee -a /etc/apt/sources.list.d/signal-xenial.list
	$ sudo apt update && sudo apt install signal-desktop

2. I tried to follow https://support.signal.org/hc/en-us/articles/115002502511-How-do-I-migrate-messages-to-the-new-Signal-Desktop-

3. I opened the old chromium signal app, there was a message saying it needed to be updated. Clicking on that message opened https://chrome.google.com/webstore/detail/signal-private-messenger/bikioccmkafdpakkkcpdbppfkghcmihk I was confused that there was no "update" button or similar. I tried several times, closing and opening both the chromium browser and the app, until at some point the "you need to update" message was gone and it apparently also was the current version: it included the " Migrate to standalone" menu item. 

4. In this updated chromium app there were weird messages, which I then thought to just be old messages, but have now found out to be not messages between me (J) and my friends but between A and A's friends. So on the right hand side of each conversation there are messages which A has written, and on the left the messages of her friends. There is also a chat with my name, which contains the messages between us, hers on the right, mine on the left. The names of the contacts and contact photos are also A's. These are messages between March 2017 and June 2017 and also some few messages from the day before yesterday (2017-10-30)

5.  I clicked "Migrate to standalone", and imported the generated folder in the new signal app.

6. From the new desktop app I wrote "test" to a common friend (K). On his phone he received that message twice: at 11:55 in the chat with me, and at 12:03 in the chat with A. 

7. Since K's reply only reached my phone but not my the desktop app, I wanted to reconnect the desktop app to the phone. To this end I deleted my laptop under "linked devices" in the menu of the signal app on the phone. Shortly after this I did receive Ks reply at my laptop.

8. I found out that the messages were not old messages of mine, but As messages (see description in step 4) 

9. Shock, horror and confusion

Only possible hint for an explanation I have that in 2015 or so A gave me an old phone of hers (not the one I'm using right now!) and on that phone first she and later I both used signal. I don't know exactly what degree of "wiping the phone" we did in context of the transfer.

My phone:
Device  : OnePlus A0001 (bacon)
Android : 6.0.1 (2b5f277d1c, MHC19Q)
Memory  : 73M (4.69% free, 512M max)
Memclass: 192
OS Host : cyanogen
App     : Signal 4.11.5

I can also add log files later if that makes sense, but before posting those, I would want to at least roughly read them, for which I am too tired right now. Posting this anyway because it seems pretty major.

**Comment by moxie0** [comment]
@Janila Seems like this is a Signal-Desktop thing? Did you mean to file this in the Signal-Android issue tracker?

**Comment by Janila** [comment]
You're probably right, so I reposted it at https://github.com/WhisperSystems/Signal-Desktop/issues/1673

**Comment by Natanji** [comment]
@moxie0 Could you still look into it? The explanations for this behaviour, so far, rely on assumptions about what the Android version does. A theory seems to be that somehow, using a device that another Signal user had used in the past could somehow lead to the old private keys to remain on the device, which on key generation of Signal-Desktop it would also link against then. Is (or was, in older versions) this in any way possible? It seems unlikely, but then again nobody understands this behaviour at this point.

Signal-Android is also affected by this because we've confirmed that there are right now two Desktop devices able to send+receive messages from A's account, while A's Android lists only a single device as linked. That seems like a Android security-relevant bug. Under which circumstances is this possible? Does this prove that the two devices use the same crypto keys, or is there a possibility that there are two additional keys but only one is shown in Android? The latter, I would consider a bug in the Android version.

**Comment by Janila** [comment]
Mystery solved, no security problem. See other bug report :)
